process.c

```diff
@@ -4748,6 +4750,8 @@ rb_f_system(int argc, VALUE *argv, VALUE _)
 
     if (pid > 0) {
         VALUE status = rb_process_status_wait(pid, 0);
+
+        rb_check_typeddata(status, &rb_process_status_type);
         struct rb_process_status *data = RTYPEDDATA_DATA(status);
```

Comment:
Ditto.

```suggestion
        struct rb_process_status *data = rb_check_typeddata(status, &rb_process_status_type);
```

And `NIL_P` check is not needed here?

Reply:
That is a good question. My understanding is that `rb_f_system` always blocks, so `nil` cannot be returned. `nil` is only returned for non-blocking `Process#wait` IIRC.